### ISSUE TITLE
WakeUp: Invalid argument (os error 22)

### ISSUE THREAD

**Issue by COhsrt** [PoC-Consortium/scavenger#7]
Running scavenger (latest master) on debian stretch in an non-root account called "miner". HDD Wakeup fails there, disabled it for now.

System info:
```
$ uptime
 15:36:29 up 31 days, 13:42,  1 user,  load average: 4.90, 2.73, 2.15
$ cat /proc/cmdline
BOOT_IMAGE=/boot/vmlinuz-4.9.0-6-amd64 root=UUID=1a05597b-4ce0-47ec-a745-b0bb622f7281 ro quiet pti=off
$ cat /etc/debian_version
9.4
```

```
Aug  6 15:31:22 miner01 miner[3056]: wakeup!
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_610222096_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_190694405_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_724638739_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_419527691_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_533944334_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_305111048_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_381388810_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_572083215_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_76277762_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_648360977_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_152555524_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_495805453_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_686499858_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_228833286_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_0_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_38138881_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_114416643_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_343249929_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_457666572_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
Aug  6 15:31:22 miner01 miner[3056]: error during wakeup 17633703707761102425_266972167_38138880: Invalid argument (os error 22)
Aug  6 15:31:22 miner01 miner[3056]: #011skip one round
```

**Comment by COhsrt** [comment]
Sample Mountflags of a Mining hdd:
```
$ mount |grep burst
/dev/sdl1 on /burst/10TB-0 type ext4 (ro,noatime,data=ordered)
```

**Comment by JohnnyFFM** [comment]
we will check, invalid argument --> 99% a direct-io issue. I guess we have to align the address we seek to, will be a quick fix.

**Comment by JohnnyFFM** [comment]
Hi, just fixed a bug in random scoop generation, could you please let me know if it is working now? Thx!

**Comment by AlexisEvo** [comment]
Can't speak for OP, but your commit resolved the error on my miners.

**Comment by COhsrt** [comment]
fixed with e4dcf79

```
Aug  6 17:52:11 miner01 miner[15495]: round finished after 30371ms
Aug  6 17:53:01 miner01 miner[15495]: new block: height: 520287, scoop: 1146
Aug  6 17:53:03 miner01 miner[15495]: nonce 727024754 -> deadline: 137382
Aug  6 17:53:28 miner01 miner[15495]: nonce 454665251 -> deadline: 25518
Aug  6 17:53:32 miner01 miner[15495]: round finished after 30761ms
Aug  6 17:53:44 miner01 miner[15495]: new block: height: 520288, scoop: 886
Aug  6 17:54:17 miner01 miner[15495]: round finished after 32291ms
Aug  6 17:58:17 miner01 miner[15495]: wakeup!
Aug  6 18:01:08 miner01 miner[15495]: new block: height: 520289, scoop: 3585
Aug  6 18:01:18 miner01 miner[15495]: nonce 48832975 -> deadline: 20253
Aug  6 18:01:39 miner01 miner[15495]: round finished after 31055ms
```
